lib/liquid/tags/include.rb

```diff
@@ -45,7 +44,13 @@ def parse(tokens)
     def render(context)
       partial = load_cached_partial(context)
       variable = context.evaluate(@variable_name)
-
+      
+      if @template_name =~ QuotedString
+          context_variable_name = @template_name[1..-2].split('/'.freeze).last
+      else
+          context_variable_name = context[@template_name]
```

Comment:
The issue with this is that it doesn't allow the template name to come from a variable.